### PR TITLE
Separate ZSS from app-server into its own component

### PR DESCRIPTION
**Pull request by sakeerthy**
Signed-off-by: Suneeth Keerthy <skeerthy@rocketsoftware.com>

Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [ ] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [ ] Feature 
- [ ] Other... Please describe:

#### Relevant issues

Fixes 

#### Changes proposed in this PR

- 

#### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number:

### CODE REVIEW

File: scripts/zowe-install-zlux.sh [hunk 1]
```diff
@@ -22,7 +22,15 @@ echo "<zowe-install-zlux.sh>" >> $LOG_FILE
...
 umask 0002
 APP_SERVER_COMPONENT_DIR=${ZOWE_ROOT_DIR}/components/app-server
+ZSS_COMPONENT_DIR=${ZOWE_ROOT_DIR}/components/zss
 mkdir -p ${APP_SERVER_COMPONENT_DIR}
+mkdir -p ${ZSS_COMPONENT_DIR}
+cd ${ZSS_COMPONENT_DIR}
+pax -r -px -f $INSTALL_DIR/files/zss.pax
+mkdir -p bin
+cd bin
+touch configure.sh
```
Comment (1000TurquoisePogs) [inline scripts/zowe-install-zlux.sh:32]: no need for touch here

File: scripts/zowe-install-zlux.sh [hunk 2]
```diff
@@ -22,7 +22,15 @@ echo "<zowe-install-zlux.sh>" >> $LOG_FILE
 
 umask 0002
 APP_SERVER_COMPONENT_DIR=${ZOWE_ROOT_DIR}/components/app-server
+ZSS_COMPONENT_DIR=${ZOWE_ROOT_DIR}/components/zss
 mkdir -p ${APP_SERVER_COMPONENT_DIR}
+mkdir -p ${ZSS_COMPONENT_DIR}
```
Comment (1000TurquoisePogs) [inline scripts/zowe-install-zlux.sh:27]: you DO Need this line, to make the dir

File: scripts/zowe-install-zlux.sh [hunk 3]
```diff
@@ -22,7 +22,15 @@ echo "<zowe-install-zlux.sh>" >> $LOG_FILE
...
+cd ${ZSS_COMPONENT_DIR}
+pax -r -px -f $INSTALL_DIR/files/zss.pax
```
Comment (1000TurquoisePogs) [inline scripts/zowe-install-zlux.sh:29]: you dont need these lines, because further down is the correct stuff

File: scripts/zowe-install-zlux.sh [hunk 4]
```diff
@@ -55,9 +63,11 @@ cp -f ${INSTALL_DIR}/files/zlux/config/plugins/* zlux-app-server/defaults/plugin
 
 echo "Unpax zssServer " >> $LOG_FILE
 cd zlux-app-server
-pax -r -px -f $INSTALL_DIR/files/zss.pax bin
+pax -r -px -f $INSTALL_DIR/files/zss.pax ${ZSS_COMPONENT_DIR}
```
Comment (1000TurquoisePogs) [inline scripts/zowe-install-zlux.sh:66]: need to test to confirm if this places samplib/loadlib in a duplicate spot or if that is the intended location

File: scripts/zowe-install-zlux.sh [hunk 5]
```diff
@@ -55,9 +63,11 @@ cp -f ${INSTALL_DIR}/files/zlux/config/plugins/* zlux-app-server/defaults/plugin
...
 cd bin
-extattr +p zssServer
+# extattr +p zssServer
```
Comment (1000TurquoisePogs) [inline scripts/zowe-install-zlux.sh:68]: comments not needed anymore, but, chmod +x may or may not be needed for the contents of the bin folder.
also, you may need to chmod 750 of app-server/share/zlux-app-server/defaults/serverConfig
and chmod 740 of app-server/share/zlux-app-server/defaults/serverConfig/*

File: scripts/zowe-install-zlux.sh [hunk 6]
```diff
@@ -54,11 +56,11 @@ cp -f ${INSTALL_DIR}/files/zlux/config/zluxserver.json zlux-app-server/defaults/
...
 
 echo "Unpax zssServer " >> $LOG_FILE
-cd zlux-app-server
+cd ${ZSS_COMPONENT_DIR}
 pax -r -px -f $INSTALL_DIR/files/zss.pax bin
-cd bin
-extattr +p zssServer
-cd ../..
+extattr +p bin/zssServer
+cp -r ${ZSS_COMPONENT_DIR}/bin/z* ${APP_SERVER_COMPONENT_DIR}/share/zlux-app-server/bin
```
Comment (1000TurquoisePogs) [inline scripts/zowe-install-zlux.sh:62]: oh one more thing... extattr +p is not portable. Copies lose it, so you need `extattr +p ${APP_SERVER_COMPONENT_DIR}/share/zlux-app-server/bin/zssServer`

**Comment by 1000TurquoisePogs** [comment]
Tested and should be harmless because (for now) we're making a duplicate of the zss content in case people were trying to hardcode its location (even though they shouldnt!)
